### PR TITLE
[16.0][IMP] stock_move_auto_assign: reassing on move cancellation

### PR DESCRIPTION
**Pull request by lmignon**
* [x] includes #1531

### CODE REVIEW

**Comment by lmignon** [comment]
cc @rousseldenis @guewen 
see https://github.com/OCA/stock-logistics-warehouse/pull/1532/commits/e0465177125ca6f25ef8f0736238e7bb365b2692

**Comment by lmignon** [comment]
cc @sebalix @jbaudoux

**Review by adrienpeiffer — APPROVED** [review]
Code review

**Comment by adrienpeiffer** [comment]
@rousseldenis Could you add your functional review ?

File: stock_move_auto_assign/models/stock_move.py [hunk 1]
```diff
@@ -13,10 +14,29 @@ class StockMove(models.Model):
...
+    def _action_cancel(self):
+        moves_with_reservation_to_cancel = self.filtered(
+            lambda m: m.state not in ("done", "cancel")
+            and float_compare(
+                m.reserved_availability, 0, precision_rounding=m.product_uom.rounding
+            )
+            > 0
+        )
+        res = super()._action_cancel()
+        moves_with_reservation_cancelled = (
```
Comment (jbaudoux) [inline stock_move_auto_assign/models/stock_move.py]: we should favor US english
```suggestion
        moves_with_reservation_canceled = (
```
same below

File: stock_move_auto_assign/tests/test_auto_assign.py [hunk 2]
```diff
@@ -43,6 +43,35 @@ def test_move_done_enqueue_job(self):
             self.assertEqual(delay_args, (self.shelf1_loc | self.shelf2_loc,))
             self.assertDictEqual(delay_kwargs, {})
 
+    def test_move_cancelled_with_reservation_enqueue_job(self):
```
Comment (jbaudoux) [inline stock_move_auto_assign/tests/test_auto_assign.py]:
```suggestion
    def test_move_canceled_with_reservation_enqueue_job(self):
```
same below

**Review by jbaudoux — COMMENTED** [review]
LG, not tested

**Review by jbaudoux — APPROVED** [review]

File: stock_move_auto_assign/tests/test_auto_assign.py (hunk 2 above)
Comment (lmignon) [reply stock_move_auto_assign/tests/test_auto_assign.py]: done

File: stock_move_auto_assign/models/stock_move.py (hunk 1 above)
Comment (lmignon) [reply stock_move_auto_assign/models/stock_move.py]: done

**Review by rousseldenis — APPROVED** [review]
Functional review

**Pull request merged**